prometheus-stats

```diff
@@ -74,7 +74,32 @@ queue_time_wait_seconds_bucket{{le="3600"}} {test_runs_wait_1h}
 queue_time_wait_seconds_bucket{{le="+Inf"}} {test_runs}
 queue_time_wait_seconds_sum {test_runs_queue_wait_sum}
 queue_time_wait_seconds_count {test_runs}
+
+"""
+
+    # failures
+    top_failures = cursor.execute("""
+            SELECT TestRuns.project, t1.testname, (t2.failed * 100.0) / COUNT(run) AS percent
+            FROM Tests AS t1
+            JOIN TestRuns ON t1.run = TestRuns.id
+            JOIN (
+                SELECT testname, COUNT(run) as failed
+                FROM Tests
+                JOIN TestRuns ON Tests.run = TestRuns.id
+                WHERE TestRuns.time > strftime('%s', 'now', '-7 days') AND Tests.failed = 1
+                GROUP BY testname
+            ) AS t2 ON t1.testname = t2.testname
+            WHERE TestRuns.time > strftime('%s', 'now', '-7 days') AND t2.failed > 1
+            GROUP BY t1.testname
+            ORDER BY percent DESC
```

Comment:
cannot we also do `WHERE percent > 10`? and then we don't need to limit it to 50 and subsequently check if failure rate is over 10%.

Or we cannot use `percent` in WHERE clause?

Reply:
I tried that, but unfortunately WHERE can't be used with aggregates (`SUM`). `HAVING` only works for aggregates. I did not find a way to do this in one select. Possibly we could wrap this into yet another sub-select, do the full table first, and then only select the ones which are > 10%. I'll try that in a bit.

Reply:
Right, was afraid of that.
If it will take another subquery, lets maybe not do that as it will most likely become too difficult to read and maintain.